### ISSUE TITLE
Is there a way to define many to many (or any other than one to one) relationships?

### ISSUE THREAD

**Issue by teeli** [darrylkuhn/dialect#15]
Let's say I have table called employees, something like this:

```
CREATE TABLE employees
(
   id serial NOT NULL,
   name character varying(255) NOT NULL
)
```

With data like:
`INSERT INTO employees (id, name) VALUES (1, 'john doe')`
`INSERT INTO employees (id, name) VALUES (2, 'john smith')`

And a table called projects, something like this:

```
CREATE TABLE projects
(
   id serial NOT NULL,
   name character varying(255) NOT NULL,
   employees jsonb NOT NULL
)
```

Where employees is an array of employee IDs. Something like:
`INSERT INTO projects (name, employees) VALUES ('cool new project', '[{employee_id: 1}, {employee_id: 2}]')`

Is there a way to define a many to many relation in a similar way to one to one relation that would automatically load something like this using `Project::with('projectEmployees')->find(1);`;

```
{
  "id": 1,
  "name": "cool new project",
  "employees": [
    {"id": 1, "name" :"john doe"},
    {"id": 2, "name": "john smith"}
  ]
}
```

So basically, I'm trying replace a separate pivot table by using JSON data instead in the hope to keep database structure more simple and to have more flexible way to define relationship in a more NoSQL fashion. Do tell me if this is a bad idea to begin with, since I'm pretty new to Postgres and JSON datatypes and I've only really worked with MySQL and MongoDB before.

**Comment by darrylkuhn** [comment]
Don't think the package would support this as is, and I think it would be fairly difficult... What you're describing is a [many-to-many relationship](https://laravel.com/docs/5.1/eloquent-relationships#many-to-many). The many-to-many relation in laravel expects a join table so to acomplish this you'd need another relation type or you'd need to hack up the existing [BelongsToMany](https://laravel.com/api/5.1/Illuminate/Database/Eloquent/Relations/BelongsToMany.html) class. On the other hand I am having trouble envisioning the use case... no schema data structures are great when a single set of records will "look different" so in this case you'd be saying:

> The way I relate projects to employees is different on a case by case basis

Is that true? If not then I don't think you get much by storing this kind of data in a json structure. If so then what do the other cases look like and how will your model (code) know when to construct the relation one way v.s. another?

**Comment by teeli** [comment]
Hmm, it's kind of different I guess. I was thinking that would be a nifty NoSQL-ish way to avoid pivot tables when the amount of employees assigned to a project might vary. 

As I mentioned, I'm more experience with MySQL and on the other hand NoSQL DBs like MongoDB and was thinking Postgres JSON fields might offer some NoSQL-ish sugar on top of traditional SQL databases. Now that I've dived a little deeper into this topic, I agree that my initial idea is flawed and I should probably handle the relationships the traditional way, that is one of the things where SQL excels after all and just use the JSON fields to only store schemaless data.

**Comment by darrylkuhn** [comment]
Awesome - good luck and let me know if you need any more help with the package.
